src/server/macos.rs

```diff
@@ -334,17 +334,28 @@ impl<'os> Method for PackageMethod<'os, Macos> {
         unix::upgrade(todo, options, self)
     }
     fn destroy(&self, options: &Destroy) -> anyhow::Result<()> {
-        let mut found = false;
-        log::info!(target: "edgedb::server::destroy",
-            "Unloading service");
-        let unit_path = unit_path(&options.name)?;
-        process::run(&mut StdCommand::new("launchctl")
-            .arg("unload").arg(&unit_path))?;
-        if unit_path.exists() {
+        let mut found;
+        let mut unit_file = unit_path(&options.name, true)?;
+        if unit_file.exists() {
             found = true;
+        } else {
+            unit_file = unit_path(&options.name, false)?;
+            found = unit_file.exists();
+        }
+        if found {
+            match launchctl_status(&options.name, false, &StatusCache::new()) {
```

Comment:
I feel like I should do something to these `launchctl_status` calls, but I'm not sure about the direction. Please kindly advise.

Reply:
Have you tested without them? Usually supervisors work like we want: don't do anything (i.e. no error) if service is already running. So are you sure that these status calls are required?

If you are, it looks like you always only look for `Running`. So you might wrap it into `is_running` function.

Reply:
Right - will do, thanks!

Reply:
Hmm I found a failing situation where the server process exited normally (e.g. killed with SIGTERM) but the service was never unloaded (pid = `-`, `launchctl_status() == Failed`), in which case the service can be restarted with `kickstart`, but `bootstrap` will fail because the service is already loaded. So I'll keep using `launchctl_status()` in `start` (`kickstart` if `Failed` or `Error`, `bootstrap` if `Inactive`, log info if `Running`), and use `fn is_service_loaded() { launchctl_status() != Inactive }` in `stop`/`destroy`/`status`.

Reply:
Maybe we need to use `kill` for stop and `bootout` only on destroy? So that start is `kickstart` (except for manually started, which we `bootstrap` on start, but I don't think we have to `bootout` them on stop).

So we only need to check for status for start of manually started instances.
(hope this makes sense)

Reply:
Oops sorry I missed this comment. Yes, that's actually much cleaner to me - `bootstrap` and `bootout` for loading/unloading the service with launchd in only `init` and `destroy`, and `kickstart` / `SIGTERM` for (re-)starting and stopping the registered service. That way, we don't need to keep the plist file for manually-started services as you mentioned - because we only need to bootstrap them into launchd once at `init`.